### PR TITLE
More code cleanup

### PR DESCRIPTION
**Pull request by BenTheElder**
- add godoc to the new nodes package 
- use the Node type in more places

/assign @amwat

### CODE REVIEW

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -367,6 +367,6 @@ func stringSliceToByteSliceSlice(ss []string) [][]byte {
 }
 
 // ListNodes returns the list of container IDs for the "nodes" in the cluster
-func (c *Context) ListNodes() (containerIDs []string, err error) {
+func (c *Context) ListNodes() ([]*nodes.Node, error) {
```
Comment (munnerz) [inline pkg/cluster/cluster.go]: (nit) is it useful/does it make sense for this to be a slice of pointers, as opposed to `[]nodes.Node`?

**Comment by munnerz** [comment]
Looks good to me!

/lgtm
/hold

File: pkg/cluster/cluster.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/cluster/cluster.go]: This plays nicer with Delete(), so I'm on the fence about what to do here.

Comment (munnerz) [reply pkg/cluster/cluster.go]: SGTM. I don't think it's a big issue really 😄 

Feel free to `/hold cancel`!

Comment (BenTheElder) [reply pkg/cluster/cluster.go]: This actually annoys me but I haven't thought of a better solution yet so I'll just merge this for now 😅

**Pull request merged**